bip-0360.mediawiki

```diff
@@ -0,0 +1,637 @@
+
+=== Transaction ID ===
+
+The transaction ID is computed as the HASH256 of the serialized transaction, including the attestation and witness
+(if a witness is present). When decoded, this is called the qtxid, which will differ from the txid and wtxid if an
+attestation is present.
+
+=== Attestation Structure ===
+
+The attestation field consists of:
+
+* <code>num_pubkeys</code>: The number of public keys ([https://learnmeabitcoin.com/technical/general/compact-size/ compact size]).
+
+For each public key:
+
+* <code>pubkey_length</code>: compact size length of the public key.
+* <code>pubkey</code>: The public key bytes.
+
+Then:
+
+* <code>num_signatures</code>: The number of signatures (compact size).
+
+For each signature:
+
+* <code>signature_length</code>: compact size length of the signature.
+* <code>signature</code>: The signature bytes.
+
+This structure repeats for each input, in order, for flexibility in supporting multisig schemes and various
+quantum-resistant algorithms.
+
+For each input, a separate attestation field is used. To know how many attestation fields are present, implementations
+must count the number of inputs present in the transaction.
+
+=== Signature Algorithms ===
+
+The specific quantum-resistant signature algorithm used is inferred from the length of the public key.
+Implementations must recognize the supported algorithms and validate accordingly.
+
+Supported PQC algorithms and their NIST Level V parameters:
+
+* '''SPHINCS+-256f:'''
+  * Public Key Length: 64 bytes
+  * Signature Length: 49,856 bytes
+* '''CRYSTALS-Dilithium Level 5:'''
+  * Public Key Length: 2,592 bytes
+  * Signature Length: 4,595 bytes
+* '''FALCON-1024:'''
+  * Public Key Length: 1,793 bytes
+  * Signature Length: 1,280 bytes
+
+Implementations must reject public keys and signatures that do not match expected lengths for supported algorithms.
+
+If a new algorithm is added, and one of the byte sizes overlaps, then an additional byte should be prepended to the
+new algorithm's public key length that indicates the specific algorithm used.
+
+=== Script Validation ===
+
+To spend a P2QRH output, the following conditions must be met:
+
+1. The <code>scriptPubKey</code> must be of the form:
+
+OP_PUSHNUM_3 <32-byte hash>
+
+2. The attestation must include:
+
+* The quantum-resistant public key(s) whose HASH256 concatenated and hashed again matches the <nowiki><hash></nowiki> in
+the <code>scriptPubKey</code>.
+
+* Valid signatures corresponding to the public key(s) and the transaction data.
+
+3. For multi-signature schemes, all required public keys and signatures must be provided for that input within the
+attestation. Public keys that are not needed can be excluded by including their hash in the attestation accompanied
+with an empty signature. This includes classical Schnorr signatures.
+
+==== Sighash Calculation ====
+
+The sighash for P2QRH outputs follows the same procedure as defined in BIP-0143 for SegWit transactions:
```

Comment:
Is there any reason why this is based on SegWit's BIP-0143 instead of Taproot which brought some reasonable changes? See https://github.com/bitcoin/bips/blob/master/bip-0341.mediawiki#signature-validation-rules

Reply:
That's a good catch, I hadn't thought of that. I'll get that updated.